### PR TITLE
Add clang-format

### PR DESCRIPTION
**Pull request by Trompettesib**
(good luck :p)

The only real change is in Dimension.cpp
We removed the task object creation from the structure creation

### CODE REVIEW

**Review by huntears — DISMISSED** [review]
xd

File: cubic-server/Client.hpp [hunk 1]
```diff
@@ -2,45 +2,62 @@
...
+#define DEFAULT_FAVICON                                                                                                                                                         \
```
Comment (NekoSelena) [inline cubic-server/Client.hpp:40]: Mamma mia

**Review by NekoSelena — APPROVED** [review]

**Review by PierrickMarguin — APPROVED** [review]

**Review by Thomas-Harit — APPROVED** [review]

File: cubic-server/logging/Logger.hpp [hunk 2]
```diff
@@ -24,129 +24,118 @@
 #define _LOG8(msg, ...) msg << _LOG7(__VA_ARGS__)
 #define _LOG9(msg, ...) msg << _LOG8(__VA_ARGS__)
 
-#define _LOGN(...) \
```
Comment (STMiki) [inline cubic-server/logging/Logger.hpp:27]: OOF

File: cubic-server/logging/Logger.hpp [hunk 3]
```diff
@@ -24,129 +24,118 @@
...
+
+/**
+ * @brief Handles logging in a file.
+ */
+class Logger {
+public:
+    static Logger *get_instance();
+
+    ~Logger();
+    ;
```
Comment (STMiki) [inline cubic-server/logging/Logger.hpp:90]: ?

File: cubic-server/protocol/ClientPackets.cpp [hunk 4]
```diff
@@ -187,12 +161,7 @@ std::shared_ptr<std::vector<uint8_t>> protocol::createChunkDataAndLightUpdate(co
 std::shared_ptr<std::vector<uint8_t>> protocol::createWorldEvent(const WorldEvent &in)
 {
     std::vector<uint8_t> payload;
-    serialize(payload,
-        in.event, addInt,
-        in.position, addPosition,
-        in.data, addInt,
-        in.disableRelativeVolume, addBoolean
-    );
+    serialize(payload, in.event, addInt, in.position, addPosition, in.data, addInt, in.disableRelativeVolume, addBoolean);
```
Comment (STMiki) [inline cubic-server/protocol/ClientPackets.cpp:164]: oof²

**Review by STMiki — APPROVED** [review]
I am not a fan of wrapping one line function, but for most of the function I can see the advantage so this is good overall
```cpp
void fn() { do_something(); }
```

File: cubic-server/protocol/ClientPackets.cpp (hunk 4 above)
Comment (huntears) [reply cubic-server/protocol/ClientPackets.cpp:164]: Some code would need to be formatted by hand and then deactivate the clang-format on this temporally.

File: cubic-server/logging/Logger.hpp (hunk 3 above)
Comment (Trompettesib) [reply cubic-server/logging/Logger.hpp:90]: This is cringe

File: cubic-server/concept.hpp [hunk 5]
```diff
@@ -13,7 +13,7 @@ requires(std::ostream &os, const T &value) {
 
 // clang-format on
 
-template<typename T>
+template <typename T>
```
Comment (STMiki) [inline cubic-server/concept.hpp:16]: Why ?

Comment (Trompettesib) [reply cubic-server/concept.hpp:16]: I don't know

**Review by huntears — APPROVED** [review]

**Comment by huntears** [comment]
Merging rn as the changes were accepted before, there are some problems with it but nothing we can't fix later.

**Pull request merged**